src/component/component_node.ts

```diff
@@ -124,7 +124,7 @@ export class ComponentNode<T extends typeof Component = any> implements VNode<Co
     try {
       await Promise.all(this.willStart.map((f) => f.call(component)));
     } catch (e) {
-      handleError(this, e);
+      handleError(this, e as Error);
```

Comment:
I don't think that's correct. Either error handlers should handle non-error objects (willStart can throw anything), or we should wrap non-error objects somewhere into real errors. (IMO the error handlers should just not expect to get a real error, let them deal with a string if willStart throws a string)

(this is not a critique of this change, but of the API of error handlers)